src/dune_engine/report_errors_config.ml

```diff
@@ -0,0 +1,17 @@
+(** Controls when the errors are reported.
+
+    - [Early] - report errors as soon as they are discovered. This means the
+      ordering of errors can depend on build scheduling.
+    - [Deterministic] - report errors at the end of the build in a deterministic
+      order.
+    - [Twice] - report each error twice: once as soon as the error is discovered
+      and then again at the end of the build, in a deterministic order. *)
+type t =
+  | Early
+  | Deterministic
+  | Twice
+
+let default : t =
+  match Dune_util.Config.inside_dune with
```

Comment:
It might make sense to default to `Deterministic` when `inside_ci` is `true`.

Reply:
I was thinking in some CIs it's possible to inspect partial output (e.g. I beleive our old Windows CI let you do this, less sure about current ones), which can be useful.
It can also be useful if dune is killed by timeout with no chance of printing its errors.
In general I'm thinking people usually don't care about CI logs being deterministic, only about the artifacts and the CI status being that way.